sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceClient.java

```diff
@@ -355,7 +428,28 @@ public PagedIterable<BlobContainerItem> listBlobContainers() {
      */
     @ServiceMethod(returns = ReturnType.COLLECTION)
     public PagedIterable<BlobContainerItem> listBlobContainers(ListBlobContainersOptions options, Duration timeout) {
-        return new PagedIterable<>(blobServiceAsyncClient.listBlobContainersWithOptionalTimeout(options, timeout));
+        throwOnAnonymousAccess();
+        BiFunction<String, Integer, PagedResponse<BlobContainerItem>> pageRetriever = (marker, pageSize) -> {
+            ListBlobContainersOptions finalOptions = options != null ? options : new ListBlobContainersOptions();
+
+            if (pageSize != null) {
+                finalOptions.setMaxResultsPerPage(pageSize);
```

Comment:
This will end up mutating the options passed by the caller, may need to copy the `options` so we don't override an external object. Or, have `listBlobContainersSegment` take the splayed out options from `ListBlobContainersOptions` instead of the options object itself.

```java
listBlobContainersSegment(String marker, String prefix, Integer maxResultsPerPage, Duration timeout)
```